src/configprovider.ts

```diff
@@ -4,23 +4,22 @@ import { getMesonTargets } from "./introspection";
 import { Target } from "./types";
 import { extensionConfiguration, getTargetName } from "./utils";
 
-export enum MIModes {
-  lldb = "lldb",
-  gdb = "gdb",
-}
-
-export class DebugConfigurationProvider implements vscode.DebugConfigurationProvider {
-  private path: string;
+export abstract class MesonDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+  private readonly path: string;
 
   constructor(path: string) {
     this.path = path;
   }
 
+  abstract getName(): string;
+  abstract createDebugConfiguration(target: Target): Promise<vscode.DebugConfiguration>;
+
   async createBaseDebugConfiguration(target: Target): Promise<vscode.DebugConfiguration> {
```

Comment:
I would change this function name to `createDebugConfiguration`. Make it not abstract, and use super in the Cppdbg class to access it after overriding it. Does that make sense?